apps/frontend/src/app/Database/DataManager.ts

```diff
@@ -73,14 +73,36 @@ export class DataManager<CacheKey extends string, GOKey extends string, CacheVal
     this.listeners[key]?.forEach(cb => cb(key, reason, object))
     this.anyListeners.forEach(cb => cb(key, reason, object))
   }
-  remove(key: CacheKey) {
+  remove(key: CacheKey, notify = true) {
     const rem = this.data[key]
     delete this.data[key]
     this.removeStorageEntry(key)
 
-    this.trigger(key, "remove", rem)
+    if (notify) this.trigger(key, "remove", rem)
     delete this.listeners[key]
   }
+  swapId(oldKey: CacheKey, newKey: CacheKey, notify = false): boolean {
+    if (oldKey === newKey) return false
+    const value = this.get(oldKey)
+    if (!value) return false
+    if (this.get(newKey)) return false
```

Comment:
Why fail if the artifact exists? Should this renamed to `changeId` since it doesn't look like there is any swapping